File: financial-tracebility/chaincode-go/chaincode/context.go
package chaincode

import "github.com/hyperledger/fabric-contract-api-go/contractapi"

type TransactionContextInterface interface {
	contractapi.TransactionContextInterface
	GetOrderList() *OrderList
}

type TransactionContext struct {
	contractapi.TransactionContext
	orderList *OrderList
}

func (tc *TransactionContext) GetOrderList() *OrderList {
	if tc.orderList == nil {
		tc.orderList = newOrderList(tc)
	}
	return tc.orderList
}

File: financial-tracebility/chaincode-java/src/main/java/chaincode/java/Context.java
package chaincode.java;

import org.hyperledger.fabric.shim.ChaincodeStub;

public class Context extends org.hyperledger.fabric.contract.Context {

    private OrderList orderList;

    Context(ChaincodeStub stub) {
        super(stub);
        this.orderList = new OrderList(this);
    }

    public OrderList getOrderList() {
        return this.orderList;
    }
}
